test/simulator/moves/encore.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe('Encore', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should not affect Focus Punch if the the user\'s decision is not changed', function () {
+		battle = common.createBattle({gameType: 'doubles'}, [
+			[
+				{species: "Smeargle", level: 50, ability: 'owntempo', moves: ['splash', 'focuspunch']},
+				{species: "Abra", level: 1, ability: 'innerfocus', moves: ['knockoff', 'teleport']},
+			],
+			[
+				{species: "Smeargle", ability: 'owntempo', item: 'laggingtail', moves: ['encore', 'splash']},
+				{species: "Zigzagoon", level: 1, ability: 'pickup', moves: ['extremespeed']},
+			],
+		]);
+
+		// If the Focus Punch user is not interrupted the attack is expected to be successful.
+		battle.makeChoices('move focuspunch 1, move knockoff 1', 'move splash, move extremespeed 2');
+		const hp = battle.p2.active[0].hp;
+		assert.notStrictEqual(hp, battle.p2.active[0].maxhp);
+
+		// If a user's previous move was Focus Punch and it is Encored into Focus Punch while attempting to
+		// execute the move, the regular "you can't be hit" effect for Focus Punch will be enforced.
+		battle.makeChoices('move focuspunch 1, move teleport', 'move encore 1, move extremespeed 1');
+		assert.strictEqual(hp, battle.p2.active[0].hp);
```

Comment:
You'll have to make the previous turn a successful Focus Punch to get this to work, sorry about that!